source/gui/accPropServer.py

```diff
@@ -90,24 +91,53 @@ def _getPropValue(self, pIDString, dwIDStringLen, idProp):
 			to extract the HWND/idObject/idChild from the identity string.
 			Note that, while one IAccPropServer implementation can annotate
 			multiple accessible elements, it is still bound to one wx.Control.
-		@type pIDString: str
 		@param dwIDStringLen: Specifies the length of the identity string specified by the pIDString parameter.
-		@type dwIDStringLen: int
-		@param idProp: Specifies a GUID indicating the desired property.
-		@type idProp: One of the oleacc.PROPID_* GUIDS
-		@return A tuple of the out params for the `IAccPropServer::GetPropValue` method: `VARIANT* pvarValue` and `BOOL*
-		pfHasProp`. When the pfHasProp part is FALSE / self.DOES_NOT_HAVE_PROP, then the pvarValue part must be VT_EMPTY.
-		Consider using self.NO_RETURN_VALUE instead. Returning (VT_EMPTY, HAS_PROP) IS valid, meaning the property exists
-		but is empty.
+		@param idProp: Specifies a GUID indicating the desired property. One of the values from oleacc.PROPID_*
+		@return Use L{self._hasProp} to return correct values or return None if unable to supply the property.
 		"""
 		raise NotImplementedError
 
-	def GetPropValue(self, pIDString, dwIDStringLen, idProp):
+	def _hasProp(self, value: Any) -> Optional[Tuple[BOOL, Any]]:
+		"""Constructs a tuple for the `IAccPropServer::GetPropValue` method, two elements:
+			1. `VARIANT pvarValue`
+			2. `BOOL pfHasProp` (either self.HAS_PROP or self.DOES_NOT_HAVE_PROP)"""
+		return value, self.HAS_PROP
+
+	def GetPropValue(
+			self, this, # unused "this" used to indicate to comTypes we want a low level implementation
+			pIDString: str,
+			dwIDStringLen: int,
+			idProp: GUID,
+			pvarValue,
+			pfGotProp
+	) -> int:
+		""" Exposed method to get a prop value.
+		see L{_getPropValue} for more details of args.
+		Uses a low-level approach, because comtypes tries to clear the VARIANT even though it is an out param.
+		When the pfHasProp part is FALSE / self.DOES_NOT_HAVE_PROP, then the pvarValue.vt part must be VT_EMPTY.
+		@type pvarValue: POINTER[VARIANT]
+		@type pfGotProp: POINTER[BOOL]
+		"""
+		# Set values in case we return early.
+		pfGotProp[0] = self.DOES_NOT_HAVE_PROP
+		pvarValue[0].vt = VT_EMPTY
+		try:
+			ret = self._getPropValue(pIDString, dwIDStringLen, idProp)
+			if ret is None:
+				return S_OK
+			elif len(ret) != 2:
+				raise RuntimeError("_getPropValue implementation must return None or two element tuple")
+		except Exception as e:
+			log.exception()
+			return S_OK
+		if ret[1] != self.HAS_PROP:
+			return S_OK
+		pfGotProp[0] = self.HAS_PROP
```

Comment:
I think I prefer

```suggestion
		pfGotProp.contents = self.HAS_PROP
```

instead. it looks clearer.

Reply:
These libraries are very hard to read, and the documentation sucks. Though I found some help in ctypes tests, at least there are (presumably working) examples. It seems you have to actually have to do `pfGotProp.contents.value`. For a C/C++ programmer `pfGotProp[0]` probably looks more familiar when dealing with pointers.